v2/controllers/marketplace/meterbase_controller.go

```diff
@@ -583,6 +584,17 @@ func (r *MeterBaseReconciler) generateExpectedDates(endTime time.Time, loc *time
 }
 
 func (r *MeterBaseReconciler) newMeterReport(namespace string, startTime time.Time, endTime time.Time, meterReportName string, instance *marketplacev1alpha1.MeterBase, prometheusServiceName string) *marketplacev1alpha1.MeterReport {
+	// If kubeVersion < 1.20 TokenRequest and TokenRequestProjection are beta and not assumed available, use basicAuth (https port). Logical default if unknown kubeVersion parse failure
```

Comment:
We'll need to rework and merge with all the rework for prometheus. Using 1.20 as a cut off point makes sense from the Feature Gate perspective but many servers have this enabled by default before 1.20.